python/lsst/analysis/tools/actions/plot/elements/plotInfoElement.py

```diff
@@ -0,0 +1,146 @@
+        #        text-build is separated-out from addPlotInfo in plotUtils.
+        # TO DO: figure out how to get this information
+        photocalibDataset = "None"
+        astroDataset = "None"
+
+        run = plotInfo["run"]
+        datasetsUsed = f"\nPhotoCalib: {photocalibDataset}, Astrometry: {astroDataset}"
+        tableType = f"\nTable: {plotInfo['tableName']}"
+
+        dataIdText = ""
+        if "tract" in plotInfo.keys():
+            dataIdText += f", Tract: {plotInfo['tract']}"
+        if "visit" in plotInfo.keys():
+            dataIdText += f", Visit: {plotInfo['visit']}"
+
+        bandText = ""
+        for band in plotInfo["bands"]:
+            bandText += band + ", "
+        bandsText = f", Bands: {bandText[:-2]}"
+        infoText = f"\n{run}{datasetsUsed}{tableType}{dataIdText}{bandsText}"
+
+        # Find S/N and mag keys, if present.
+        snKeys = []
+        magKeys = []
+        selectionKeys = []
+        selectionPrefix = "Selection: "
+        for key, value in plotInfo.items():
+            if "SN" in key or "S/N" in key:
+                snKeys.append(key)
+            elif "Mag" in key:
+                magKeys.append(key)
+            elif key.startswith(selectionPrefix):
+                selectionKeys.append(key)
+        # Add S/N and mag values to label, if present.
+        # TODO: Do something if there are multiple sn/mag keys. Log? Warn?
+        newline = "\n"
+        if snKeys:
+            infoText = f"{infoText}{newline if magKeys else ', '}{snKeys[0]}{plotInfo.get(snKeys[0])}"
+        if magKeys:
+            infoText = f"{infoText}, {magKeys[0]}{plotInfo.get(magKeys[0])}"
+        if selectionKeys:
+            nPrefix = len(selectionPrefix)
+            selections = ", ".join(f"{key[nPrefix:]}: {plotInfo[key]}" for key in selectionKeys)
+            infoText = f"{infoText}, Selections: {selections}"
+
+        return infoText
+
+    def __call__(self, ax: Axes, figureInfo: Mapping[str, Any], **kwargs) -> Axes:
+        return self.makeElement(ax, figureInfo, **kwargs)
+
+    def makeElement(self, ax: Axes, figureInfo: Mapping[str, Any], **kwargs) -> Axes:
+        """Draw a text box that displays useful information about the plot.
+
+        Parameters
+        ----------
+        ax : `~matplotlib.axes.Axes`
+            Axes to plot on.
+        figInfo: `dict`
+            A dictionary containing information about the figure on which the
+            element is being placed.  It should have the following keys:
+
+            ``"figsize"``
+                The width and height of the figure in inches
+                (`tuple` of two `float`s)
+            ``"dpi"``
+                The dpi of the figure (`float`)
+
+        Returns
+        -------
+        ax : `~matplotlib.axes.Axes`
+            Updated axes.
+        """
+        plotInfo = kwargs["plotInfo"]
+        plotInfoText = self._buildPlotInfoText(plotInfo)
```

Comment:
Can be replaced with a call to the imported function once/if the above code is moved back into a new `parsePlotInfo` function inside `plotUtils`.